Tutorial - Zapier surveys (#11393)
PostHog/posthog.com: contents/tutorials/zapier-surveys.md

Reviewer: As it is now, this is more of a docs page than tutorial. Instead of just copying the docs, can we walk them through setting up a real life example of a Zap + add screenshots? Feel free to focus on one of the two options instead of both. 

"In this tutorial, we'll set up a Zap to add survey responses to a Google Sheet"

---
title: How to send survey responses to Zapier

> date: 2025-05-01

Author: ```suggestion
date: 2025-05-06
```

author:
  - lior-neu-ner
showTitle: true
sidebar: Docs
tags:
  - surveys
---

> It's useful to send survey responses to [Zapier](https://zapier.com/). This way you can automatically trigger workflows in thousands of other apps, such as creating tickets in your help desk, or updating your CRM with customer feedback.

Reviewer: ```suggestion
It can be useful to send survey responses to [Zapier](https://zapier.com/). This way you can automatically trigger workflows in thousands of other apps, such as creating tickets in your help desk or updating your CRM with customer feedback.
```

> To show you how, in this tutorial we set up an example Zap to add survey responses to a Google sheet.

Reviewer: ```suggestion
To show you how, in this tutorial we set up an example Zap to add survey responses to a Google Sheet.
```

## Step 1: Create a survey

> First you'll need to create a survey. Our [docs](/docs/surveys/creating-surveys) cover how to do this, so we won't go into detail here. We also have [framework-specific tutorials](/docs/surveys/tutorials#framework-guides).

Reviewer: ```suggestion
First, you'll need to create a survey. Our [docs](/docs/surveys/creating-surveys) cover how to do this, so we won't go into detail here. We also have [framework-specific tutorials](/docs/surveys/tutorials#framework-guides).
```

> Once your survey is created, submit a few sample responses.

Author: I went back and forth on whether I should include this in more detail or not, but ultimately decided that its more useful to get straight to the point of this tutorial

Reviewer: Cool with me :)

## Step 2: Set your Zap trigger

You have two options for creating your Zap and setting its trigger:

### Option 1: Via the PostHog app in Zapier  <span class="bg-gray-accent-light dark:bg-gray-accent-dark text-gray font-semibold align-middle text-sm p-1 rounded">recommended</span>

1. In PostHog, go to the [actions tab](https://us.posthog.com/project/data-management/actions).
2. Click **New action** and then **From event or pageview**.
3. Select the **Other events** tab and select for the `survey sent` event.
4. Name your action `Survey sent action` and click **Save**.
5. Then in your Zapier dashboard, go to your [zaps](https://zapier.com/app/assets/zaps) and create a new zap.
6. Add the [PostHog app](https://zapier.com/apps/posthog/integrations/webhook) as a trigger.
7. Configure the PostHog app by setting the **Trigger event** to `Action Performed`. 

> 8. Click **Continue** and select the relevant organization and project. In the Action section, select the **Survey sent** action you created in the previous steps.

Reviewer: ```suggestion
8. Click **Continue** and select the relevant organization and project. In the Action section, select the **Survey sent action** you created in the previous steps.
```

9. Click  **Continue** and test your trigger.

<ProductScreenshot
  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2025_05_02_at_11_21_35_AM_aaca8d5934.png " 
  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2025_05_02_at_11_21_35_AM_aaca8d5934.png " 
  alt="Setting up the PostHog app as a trigger in Zapier" 
  classes="rounded"
/>

### Option 2: Via PostHog's Data pipelines

1. In PostHog, click the [Data pipeline](https://us.posthog.com/pipeline/overview) tab in the left sidebar.
2. Click the [Destinations](https://us.posthog.com/pipeline/destinations) tab.
3. Search for **Zapier** and click **+ Create**.

> 4. In the **Match events and actions** section, search for `survey sent`.

Reviewer: ```suggestion
4. In the **Match events and actions** section, search for and select `survey sent`.
```

5. Then in your Zapier dashboard, create a new Zap with a [webhook trigger](https://zapier.com/apps/webhook/integrations)
6. Set the **Trigger event** in the webhook to `Catch Hook` and press **Continue**.
7. Leave **Pick off a Child key** as blank and click **Continue**. Do not test your Zap just yet.
8. Go back to your PostHog app then add your webhook URL in PostHog under **Zapier hook path**.
9. Press **Create & Enable** and then click **Start testing**.

> 10. In the test JSON, replace the `event` and `person` keys with the below (leave the rest of the json as is, as Zapier needs the `project` key):

Reviewer: ```suggestion
10. In the test JSON, replace the `event` and `person` keys with the below (leave the rest of the JSON as is, as Zapier needs the `project` key):
```

  "$survey_response_4c3581e0-0131-4c0e-b471-9d70266e565a": 9,
  "$survey_questions": [
  {
    "id": "4c3581e0-0131-4c0e-b471-9d70266e565a",
    "index": 0,
    "question": "How likely are you to recommend us to a friend?"
  }
]
}
},
"person": {
"id": "965a371a-a1a5-4881-8b95-9747d1b0d4ac",
"properties": {
  "email": "max@posthog.com"
},
"name": "Max the hedgehog",
"url": "https://us.posthog.com/person/965a371a-a1a5-4881-8b95-9747d1b0d4ac"
},

// ...rest of the json
```

1.  Click **Test function**. You should see a `Success` banner.
2.  Go back to Zapier and press **Test trigger**. You should see the test request appear.

<ProductScreenshot
  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2025_05_02_at_11_55_03_AM_be9092dc5d.png" 
  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2025_05_02_at_11_55_03_AM_be9092dc5d.png" 
  alt="Setting up the webhook trigger in Zapier using PostHog CDP" 
  classes="rounded"
/>

## Step 3: Add the Google Sheet action

1. Create a new spreadsheet and name it something like `PostHog survey tutorial`.
2. In the first row, add these headers to your columns: `email`, `question`, `response`

> 3. Go to Zapier and add Google sheet as an action.

Reviewer: ```suggestion
3. Go to Zapier and add Google Sheet as an action.
```
